### PR TITLE
JAMES-3775 Adapt official Rspamd docker image

### PR DESCRIPTION
**Pull request by quantranhong1999**

### CODE REVIEW

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

File: third-party/rspamd/docker-compose-distributed.yml [hunk 1]
```diff
@@ -108,7 +108,7 @@ services:
       redis:
         condition: service_started
     container_name: rspamd
-    image: a16bitsysop/rspamd:3.8.4-r0-alpine3.20.0-r0
+    image: rspamd/rspamd:3.9.1
     environment:
       - REDIS=redis
```
Comment (vttranlina) [inline third-party/rspamd/docker-compose-distributed.yml:113]: Do we need to update the related environment?

Comment (quantranhong1999) [reply third-party/rspamd/docker-compose-distributed.yml:113]: Yes. Good catch.

Comment (Arsnael) [reply third-party/rspamd/docker-compose-distributed.yml:113]: RspamdHealthcheckTest actually failing yes, need to adapt

Comment (quantranhong1999) [reply third-party/rspamd/docker-compose-distributed.yml]: > RspamdHealthcheckTest actually failing yes, need to adapt

fixed

**Review by vttranlina — APPROVED** [review]

**Comment by Arsnael** [comment]
Other issues, buncxh of tests failing locally in rspamd module. Bayes per user wasn't enabled, threshold werent updated... Different score for reject as well had to put it a bit lower. I was investigating that today. Hmmm here => https://github.com/Arsnael/james-project/commit/854b2882aaa7dae88af33fc3f79d92e0d4abeb25

I still have an issue locally that I don't understand => tasks to feed ham and spam, feeds all messages despite the afterDate. Which is weird cause that does not depend of rspamd. Could not find the thing there

**Comment by quantranhong1999** [comment]
> Hmmm here => https://github.com/Arsnael/james-project/commit/854b2882aaa7dae88af33fc3f79d92e0d4abeb25

Thanks for putting this together. I will have a look.

>  Bayes per user wasn't enabled, threshold werent updated...

Is the bayes per user issue fixed in your commit?

> Different score for reject as well had to put it a bit lower. 

Hmm as I remember the Rspamd score is unstable, and varies on different machines. Likely that is why we tagged Rspamd tests as unstable.

> I still have an issue locally that I don't understand => tasks to feed ham and spam, feeds all messages despite the afterDate. Which is weird cause that does not depend of rspamd. Could not find the thing there

I think I know that one. I will try to fix it today.

**Comment by vttranlina** [comment]
> Hmm as I remember the Rspamd score is unstable, and varies on different machines. Likely that is why we tagged Rspamd tests as unstable.

Yes. similar experience

**Comment by Arsnael** [comment]
> Is the bayes per user issue fixed in your commit?

Yes. Needed to put the statistics.conf file into `local.d`, or it wasn't taken into account. Was the same for actions.conf file, or it was taking the threshold by default for reject (15.0)

**Comment by quantranhong1999** [comment]
> EDIT we might want to pay attention to that detail in our current deployments when switching too

At least the MU one looks good. All config files are put in `local.d`.

**Review by Arsnael — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

**Pull request merged**